### PR TITLE
Create JiraReader for loading Jira issues

### PR DESCRIPTION
**Pull request by BearGuy**
This is a new dataloader for everyone's favourite bug tracking software. It uses basic auth to access a custom Jira server and you provide a JQL query to fetch the issues you wish to load.

### CODE REVIEW

**Comment by jerryjliu** [comment]
sweet! taking a look soon

File: loader_hub/jira/base.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+from typing import List
+
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+
+class JiraReader(BaseReader):
```
Comment (jerryjliu) [inline loader_hub/jira/base.py:6]: nit: add docstring

File: loader_hub/jira/base.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    def __init__(self, email, api_token, server_url) -> None:
```
Comment (jerryjliu) [inline loader_hub/jira/base.py:8]: nit: let's type the argument fields 

e.g. 
```
email: str, api_token; str, sever_url: str
````

File: loader_hub/jira/base.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        from jira import JIRA
+
+        self.jira = JIRA(
+            basic_auth=(email, api_token),
+            server=f'https://{server_url}'
+        )
+
+    def load_data(self, query) -> List[Document]:
```
Comment (jerryjliu) [inline loader_hub/jira/base.py:17]: type query as str here

**Review by jerryjliu — COMMENTED** [review]
this is great @BearGuy!! a few minor comments

**Comment by BearGuy** [comment]
@jerryjliu feedback addressed!

**Comment by ajayep** [comment]
@jerryjliu .. seems like the loader is limited to the server edition... I was expecting cloud instance support similar to that of Confluence...

**Pull request merged**